app/src/main/java/io/gnosis/kouban/ui/transaction/details/TransactionDetailsViewModel.kt

```diff
@@ -45,30 +43,51 @@ class TransactionDetailsViewModel(
         }
 
     private fun Pair<Solidity.Address?, ServiceSafeTx>.toDetails(): List<Any> {
-        return listOf(
-            first!!,
-            second.tx.buildTransactionTypeView(first),
-            second.tx.to,
-            LabelDescription(
-                R.string.transaction_details_network_fees_label,
-                SpannableString(second.execInfo.fees.shiftedString(ETH_TOKEN_INFO.decimals, decimalsToDisplay = ETH_TOKEN_INFO.decimals))
-            ),
-            LabelDate(R.string.transaction_details_timestamp_label, dateInSecs = transaction.timestamp),
-            Link(transaction.executionHash!!, R.string.view_transaction_on),
-            LabelDescription(R.string.transaction_details_raw_data_label, SpannableString(second.tx.data))
-        )
+        val decodedData = decode(second.tx)
+        return mutableListOf<Any>().apply {
+            add(first!!)
+            add(second.buildTransactionTypeView(first, decodedData))
+            add(second.tx.to)
+            add(
+                LabelDescription(
+                    R.string.transaction_details_network_fees_label,
+                    SpannableString(second.execInfo.fees.shiftedString(ETH_TOKEN_INFO.decimals, decimalsToDisplay = ETH_TOKEN_INFO.decimals))
+                )
+            )
+            add(LabelDescription(R.string.transaction_details_operation_label, SpannableString(second.tx.operation.name)))
+            add(LabelDescription(R.string.transaction_details_value_label, SpannableString(second.tx.value.asDecimalString())))
+            second.txHash?.let { add(Link(it, R.string.view_transaction_on)) }
+            add(LabelDescription(R.string.transaction_details_raw_data_label, SpannableString(second.tx.data)))
+        }
     }
 
-    private fun SafeTx.buildTransactionTypeView(currentSafe: Solidity.Address?): TransactionTypeView =
+    private fun ServiceSafeTx.buildTransactionTypeView(currentSafe: Solidity.Address?, decodedData: DecodedData): TransactionTypeView =
         TransactionTypeView(
             currentSafe!!,
-            to,
-            transaction.type,
-            transaction.dataInfo,
-            transaction.transferInfo
+            tx.to,
+            TransactionType.Outgoing, // the only one that the backend has info for at them moment
+            decodedData.toDataInfo(tx),
+            null // Transfers
```

Comment:
do we show transfer info in details? If not I would remove last argument completely

Reply:
I added a FIXME for that too, because I don't know that for a fact. I would prefer leaving that as it is.